File: platform/test/integration/gemini-openai-sdk/transport-constant/test_case.py
#!/usr/bin/env python3
"""Direct package test for fetch_httpx_gemini_openai_constant with connection check."""

from __future__ import annotations

import argparse
import json

import re
import sys
from pathlib import Path

_HERE = Path(__file__).resolve().parent
_PLATFORM = _HERE
while _PLATFORM.name != "platform" and _PLATFORM != _PLATFORM.parent:
    _PLATFORM = _PLATFORM.parent

def _inject_paths():
    """Add polyglot/packages_py source dirs to sys.path from platform pyproject.toml."""
    toml = _PLATFORM / "pyproject.toml"
    if not toml.exists():
        return
    for line in toml.read_text().splitlines():
        m = re.search(r'path\s*=\s*"([^"]+)"', line)
        if not m:
            continue
        pkg_path = (_PLATFORM / m.group(1)).resolve()
        src = pkg_path / "src"
        sys.path.insert(0, str(src if src.is_dir() else pkg_path))

_inject_paths()

import httpx

from app_yaml_load import load_app_yaml_config
from fetch_httpx_gemini_openai_constant import (

    GEMINI_ORIGIN,
    GEMINI_CHAT_COMPLETIONS_PATH,
    GEMINI_CONNECT_TIMEOUT_S,
    GEMINI_READ_TIMEOUT_S,
    GEMINI_WRITE_TIMEOUT_S,
    GEMINI_MAX_CONNECTIONS,
    GEMINI_MAX_KEEPALIVE_CONNECTIONS,
    GEMINI_KEEPALIVE_EXPIRY_S,
    GEMINI_HTTP2_ENABLED,
)

def main() -> int:
    parser = argparse.ArgumentParser()
    parser.add_argument("--app-env", default="dev")
    parser.add_argument("--config-dir", required=True)
    args = parser.parse_args()

    app_cfg = load_app_yaml_config(config_dir=args.config_dir, app_env=args.app_env).config
    provider_config = app_cfg.get_nested("providers", "gemini_openai", default={})

    # --- Connection test (TCP reachability of GEMINI_ORIGIN) ---
    connection_test: dict = {"skipped": True, "reason": "constants-only package"}
    try:
        resp = httpx.head(GEMINI_ORIGIN, timeout=10)
        connection_test = {"skipped": False, "connected": resp.status_code < 500, "status": resp.status_code}
    except Exception as exc:
        connection_test = {"skipped": False, "connected": False, "error": str(exc)}

    summary = {
        "service": "transport-constant",
        "mode": "direct-package-import",
        "python_package": "fetch_httpx_gemini_openai_constant",
        "app_yaml_initialized": True,
        "app_yaml_providers_gemini_openai": provider_config,
        "resolved_config": {
            "gemini_origin": GEMINI_ORIGIN,
            "chat_completions_path": GEMINI_CHAT_COMPLETIONS_PATH,
            "connect_timeout_s": GEMINI_CONNECT_TIMEOUT_S,
            "read_timeout_s": GEMINI_READ_TIMEOUT_S,
            "write_timeout_s": GEMINI_WRITE_TIMEOUT_S,
            "max_connections": GEMINI_MAX_CONNECTIONS,
            "max_keepalive_connections": GEMINI_MAX_KEEPALIVE_CONNECTIONS,
            "keepalive_expiry_s": GEMINI_KEEPALIVE_EXPIRY_S,
            "http2_enabled": GEMINI_HTTP2_ENABLED,
        },
        "connection_test": connection_test,
        "uses_app_yaml_directly": False,
        "note": "fetch_httpx_gemini_openai_constant provides pre-configured constants for httpx Gemini clients.",
    }
    print(json.dumps(summary, indent=2))
    return 0

if __name__ == "__main__":
    raise SystemExit(main())

File: platform/test/integration/gemini-openai-sdk/transport-constant/test_case.mjs
#!/usr/bin/env node

import path from 'node:path';
import { pathToFileURL } from 'node:url';

const args = process.argv.slice(2);
const readArg = (name, fallback) => {
  const idx = args.indexOf(name);
  return idx >= 0 ? args[idx + 1] : fallback;
};

const appEnv = readArg('--app-env', process.env.APP_ENV || 'dev');
const configDir = readArg('--config-dir', process.env.CONFIG_DIR);
if (!configDir) {
  console.error('Missing --config-dir');
  process.exit(2);
}

const platformRoot = path.resolve(configDir, '..', '..');
const importFile = async (absPath) => import(pathToFileURL(absPath).href);

const appYamlLoadMod = await importFile(path.join(platformRoot, 'polyglot/app_yaml_load/mjs/dist/index.js'));
const constantsMod = await importFile(path.join(platformRoot, 'packages_mjs/fetch_undici_gemini_openai_constant/dist/index.js'));

const { config: appConfig } = await appYamlLoadMod.loadAppYamlConfig({ configDir, appEnv });
const providerConfig = appConfig.getNested(['providers', 'gemini_openai'], {});

// --- Connection test (TCP reachability of GEMINI_ORIGIN) ---
let connectionTest = { skipped: true, reason: 'constants-only package' };
try {
  const res = await fetch(constantsMod.GEMINI_ORIGIN, { method: 'HEAD' });
  connectionTest = { skipped: false, connected: res.status < 500, status: res.status };
} catch (err) {
  connectionTest = { skipped: false, connected: false, error: err.message };
}

const summary = {
  service: 'transport-constant',
  mode: 'direct-package-import',
  mjs_package: 'fetch-undici-gemini-openai-constant',
  app_yaml_initialized: true,
  app_yaml_providers_gemini_openai: providerConfig,
  resolved_config: {
    gemini_origin: constantsMod.GEMINI_ORIGIN,
    chat_completions_path: constantsMod.GEMINI_CHAT_COMPLETIONS_PATH,
    headers_timeout_ms: constantsMod.GEMINI_HEADERS_TIMEOUT_MS,
    body_timeout_ms: constantsMod.GEMINI_BODY_TIMEOUT_MS,
    connect_timeout_ms: constantsMod.GEMINI_CONNECT_TIMEOUT_MS,
    pool_connections: constantsMod.GEMINI_POOL_CONNECTIONS,
    pool_pipelining: constantsMod.GEMINI_POOL_PIPELINING,
    keepalive_timeout_ms: constantsMod.GEMINI_KEEPALIVE_TIMEOUT_MS,
    http2_enabled: constantsMod.GEMINI_HTTP2_ENABLED,
  },
  connection_test: connectionTest,
  uses_app_yaml_directly: false,
  note: 'fetch_undici_gemini_openai_constant provides pre-configured constants for undici Gemini clients.',
};

console.log(JSON.stringify(summary, null, 2));
